apps/ledger-live-desktop/src/renderer/families/canton/PendingTransactions/PendingTransactionDetails.tsx

```diff
@@ -0,0 +1,287 @@
+import React, { useCallback, useMemo, useState, useEffect } from "react";
+import { useTranslation } from "react-i18next";
+import { Account } from "@ledgerhq/types-live";
+import { BigNumber } from "bignumber.js";
+import Box from "~/renderer/components/Box";
+import Text from "~/renderer/components/Text";
+import Button from "~/renderer/components/Button";
+import { useAccountUnit } from "~/renderer/hooks/useAccountUnit";
+import FormattedVal from "~/renderer/components/FormattedVal";
+import IconCheck from "~/renderer/icons/Check";
+import IconCross from "~/renderer/icons/Cross";
+import { CantonAccount } from "@ledgerhq/live-common/families/canton/types";
+import {
+  OpDetailsSection,
+  OpDetailsTitle,
+  OpDetailsData,
+  B,
+  HashContainer,
+  GradientHover,
+} from "~/renderer/drawers/OperationDetails/styledComponents";
+import { SplitAddress } from "~/renderer/components/OperationsList/AddressCell";
+import CopyWithFeedback from "~/renderer/components/CopyWithFeedback";
+
+type PendingProposal = {
+  contract_id: string;
+  sender: string;
+  receiver: string;
+  amount: BigNumber;
+  memo: string;
+  expires_at_micros: number;
+  isExpired: boolean;
+};
+
+type Props = {
+  onClose?: () => void;
+  account: Account;
+  contractId: string;
+  onOpenModal: (contractId: string, action: "accept" | "reject" | "withdraw") => void;
+};
+
+const PendingTransactionDetails: React.FC<Props> = ({
+  account,
+  contractId,
+  onClose,
+  onOpenModal,
+}) => {
+  const { t } = useTranslation();
+  const unit = useAccountUnit(account);
+
+  const cantonAccount = account as CantonAccount;
+  const proposal = useMemo<PendingProposal | null>(() => {
+    const pendingTransferProposals = cantonAccount?.cantonResources?.pendingTransferProposals || [];
+    const found = pendingTransferProposals.find(p => p.contract_id === contractId);
+    if (!found) return null;
+
+    const now = Date.now();
+    const isExpired = now > found.expires_at_micros / 1000;
+    return {
+      ...found,
+      isExpired,
+      amount: new BigNumber(found.amount),
+    };
+  }, [cantonAccount, contractId]);
+
+  const handleAcceptOffer = useCallback(
+    (contractId: string) => {
+      onOpenModal(contractId, "accept");
+      // Close drawer after modal state is set
+      setTimeout(() => {
+        onClose?.();
+      }, 0);
+    },
+    [onClose, onOpenModal],
+  );
+
+  const handleRejectOffer = useCallback(
+    (contractId: string) => {
+      onOpenModal(contractId, "reject");
+      // Close drawer after modal state is set
+      setTimeout(() => {
+        onClose?.();
+      }, 0);
+    },
+    [onClose, onOpenModal],
+  );
+
+  const handleWithdrawOffer = useCallback(
+    (contractId: string) => {
+      onOpenModal(contractId, "withdraw");
+      // Close drawer after modal state is set
+      setTimeout(() => {
+        onClose?.();
+      }, 0);
+    },
+    [onClose, onOpenModal],
+  );
+
+  const formatDate = useCallback((timestamp: number) => {
```

Comment:
why do we need this function to be declared here? don't see a particular reason to "cache" this function, because declaring outside of the component achieves the same result.